File: medium_exercises/ruby_shell/exe15.rb
#! /usr/bin/env ruby

current_dir = Dir.getwd
date        = Time.new
formatted_date = "#{date.year}-#{date.month}-#{date.day}"

puts "Please enter a file extension"
file_ext = gets.chomp

puts "Please enter a new prefix"
prefix = gets.chomp

cprefix = prefix.empty? ? formatted_date : prefix

Dir.foreach(current_dir) do |file|
  if File.extname(file) == file_ext
    new_name = "#{cprefix}-#{file}"
    puts "Renaming #{file} to #{cprefix}-#{file}"
    File.rename(file, new_name)
  end
end

File: medium_exercises/python_shell/exe15.py
#!/usr/bin/env python3

from datetime import datetime
import os

date           = datetime.now()
formatted_date = date.strftime("%Y-%m-%d")
current_dir    = os.getcwd()
files          = os.listdir(current_dir)

ext = input("Please enter a file extension:")

prefix = input("Please enter a the new prefix")

cfix = prefix or formatted_date

for file in files:
    cext = os.path.splitext(file)[-1]

    if cext == ext:
        new_name = (f'{cfix}-{file}')
        print(f"Renaming: {file} to {cfix}-{file}")
        os.rename(file, new_name)


